pisa/analysis/analysis.py

```diff
@@ -2518,6 +2566,8 @@ def _minimizer_callable(self, scaled_param_vals, hypo_maker, data_dist,
         scaled_param_vals = np.where(flip_x0, 1 - scaled_param_vals, scaled_param_vals)
         # Set param values from the scaled versions the minimizer works with
         hypo_maker._set_rescaled_free_params(scaled_param_vals) # pylint: disable=protected-access
+        if hypo_maker.__class__.__name__ == "Detectors":
+            update_param_values_detector(hypo_maker, hypo_maker.params.free) #updates values for ALL detectors
```

Comment:
One more question - does `update_param_values_detector` need to be called every minimizer step? It looks like that could be rather expensive since it also makes calls back to `init_params`, and the free parameters should already be updated here.

Reply:
It is needed to update the parameters for all distribution makers